### PR TITLE
Add concepts page on AuthenticationClass and TLS server verification

### PR DESCRIPTION
**Pull request by sbernauer**
Fixes https://github.com/stackabletech/operator-rs/issues/494
Contributes to https://github.com/stackabletech/documentation/issues/311

### CODE REVIEW

File: modules/concepts/examples/authenticationclass-static-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: simple-trino-users-secret # <1>
```
Comment (maltesander) [inline modules/concepts/examples/authenticationclass-static-secret.yaml:4]: Do we want do mention trino here?  Id prefer "simple-user-credentials" or something like that. 
(Should be adapted in authenticationclass-static-authenticationclass.yaml as well)

File: modules/concepts/pages/authenticationclass.adoc [hunk 2]
```diff
@@ -0,0 +1,79 @@
+= `AuthenticationClass`
+
+`AuthenticationClass` is a CRD describing a generic authentication method like LDAP or Kerberos.
+It has `cluster` scope, so it does not belong to a specific namespace.
+Multiple operators use this CRD as a way to express the authentication of the product.
```
Comment (maltesander) [inline modules/concepts/pages/authenticationclass.adoc:5]:
```suggestion
Multiple operators use this CRD as a way to express and configure the authentication methods of their respective product.
```

File: modules/concepts/pages/authenticationclass.adoc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+At the moment the following Authentication providers are supported:
+
+* <<LDAP>>
+* TLS
```
Comment (maltesander) [inline modules/concepts/pages/authenticationclass.adoc:10]: Nothing on TLS authentication?

File: modules/concepts/pages/authenticationclass.adoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+[source,yaml]
+----
+include::example$authenticationclass-ldap-simple.yaml[]
+----
+<1> The hostname of the LDAP server without any protocol or port
+<2> The port of the LDAP server. If TLS is used it defaults to `636` otherwise to `389`
+<3> An optional searchBase where the users should be searched
+
+=== OpenLDAP
+Here is an example that is tuned for an OpenLDAP LDAP server and is configured to read bind user credentials from a secret:
```
Comment (maltesander) [inline modules/concepts/pages/authenticationclass.adoc:25]: Not sure but strange to read with the double "LADP" :D
```suggestion
Here is an example that is tuned for an OpenLDAP server and is configured to read bind user credentials from a secret:
```

File: modules/concepts/pages/tls_server_verification.adoc [hunk 5]
```diff
@@ -0,0 +1,57 @@
+= TLS server verification
+
+A TLS section is part of Stackable CRDs and describes how to connect to a TLS enabled system like LDAP or S3.
+
+If the `tls` attribute is set to `null` (or is not specified) no TLS will be used for the connection.
```
Comment (maltesander) [inline modules/concepts/pages/tls_server_verification.adoc:5]:
```suggestion
If the `tls` attribute is set to `null` (or is not specified), no TLS will be used for the connection.
```

File: modules/concepts/pages/tls_server_verification.adoc [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+A simple TLS section looks like this:
+
+[source,yaml]
+----
+include::example$tls-simple.yaml[]
+----
+
+== Verification
+The parties participating at an TLS connection can be verified using certificates.
```
Comment (maltesander) [inline modules/concepts/pages/tls_server_verification.adoc:15]:
```suggestion
The parties participating via a TLS connection can be verified using certificates.
```

File: modules/concepts/pages/tls_server_verification.adoc [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+At the moment the following verification methods are supported
```
Comment (maltesander) [inline modules/concepts/pages/tls_server_verification.adoc:16]:
```suggestion
At the moment the following verification methods are supported:
```

File: modules/concepts/pages/tls_server_verification.adoc [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+[source,yaml]
+----
+include::example$tls-server-verification-webpki.yaml[]
+----
+
+This example will use TLS and verify the server using the provided ca certificate.
+For this to work you need to create a xref:secret-operator::secretclass.adoc[] that - at least - contains the ca certificate.
+Note that a SecretClass does not need to have a key but can also work with just a ca cert.
+So if you got provided with a ca cert but don't have access to the key you can still use this method.
```
Comment (maltesander) [inline modules/concepts/pages/tls_server_verification.adoc:42]:
```suggestion
So if you were provided with a ca cert but do not have access to the key you can still use this method.
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Just minor language things. LGTM otherwise!

File: modules/concepts/examples/authenticationclass-static-secret.yaml (hunk 1 above)
Comment (sbernauer) [reply modules/concepts/examples/authenticationclass-static-secret.yaml]: Agreed, copied it from somewhere else. 64ec2ae14eb7cfde42e79ceb68195f5c11cfadac

File: modules/concepts/pages/authenticationclass.adoc (hunk 3 above)
Comment (sbernauer) [reply modules/concepts/pages/authenticationclass.adoc]: I was lazy and did get things done ^^ f571e5d2b16a70aa2766f74e529531c5175c0da7

**Review by maltesander — APPROVED** [review]
LGTM.

**Pull request merged**